apischema/deserialization/methods.py

```diff
@@ -308,6 +308,26 @@ def deserialize(self, data: Any) -> Any:
         return values
 
 
+@dataclass
+class FrozenSetMethod(DeserializationMethod):
+    constraints: Tuple[Constraint, ...]
+    value_method: DeserializationMethod
+
+    def deserialize(self, data: Any) -> Any:
```

Comment:
By the way, I think `super()` was not supported (in Cython generation) when I wrote `VariadicTupleMethod`, but this one, as well as `FrozenSetMethod` should simply inherit `ListMethod` and use for example `return tuple(super().deserialize(data))`

Reply:
After inspection, it is not straightforward, since `VariadicTupleMethod` could inherit from `ListCheckOnlyMethod` or `ListMethod`.
It looks like all collection methods could benefit from only checking their item values, so I'd say that it would be better to do some refactoring here.

Reply:
You're right, it was not a question of `super()` support … I was too quick in my response. So composition should be used the same way `VariadicTupleMethod` does.